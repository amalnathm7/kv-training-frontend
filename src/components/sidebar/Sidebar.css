.sidebar {
    position: fixed;
    top: 0;
    left: 0;
    height: 100%;
    width: 300px;
    background-color: white;
    box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
}

.item.container {
    font-size: 14px;
    color: white;
    position: absolute;
    display: flex;
    justify-content: start;
    align-items: center;
    right: 0px;
    top: 120px;
    width: 95%;
    height: 60px;
    border-top-left-radius: 30px;
    border-bottom-left-radius: 30px;
    background-color: #03AEEE;
}

.icon.container {
    margin-left: 15px;
    margin-right: 15px;
    background-color: white;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 10px;
    border-radius: 100%;
}

