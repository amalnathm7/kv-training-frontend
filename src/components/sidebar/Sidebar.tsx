import React, { useContext, useEffect, useState } from 'react';
import './Sidebar.css';
import SideBarButton from '../button/SideBarButton/SideBarButton';
import { SelectedContext, SelectedContextType } from '../../app';
import { useNavigate } from 'react-router-dom';
import { RouteConstants } from '../../constants/routeConstants';
import { useGetMyProfileQuery } from '../../services/employeeApi';
import { PermissionLevel } from '../../utils/PermissionLevel';

const Sidebar: React.FC = () => {
  const {
    data: myProfile,
    isSuccess: isMyProfileFetchSuccess,
    isError: isMyProfileFetchError
  } = useGetMyProfileQuery();
  const [isSuperAuthorized, setIsSuperAuthorized] = useState(false);
  const [isBasicAuthorized, setIsBasicAuthorized] = useState(false);
  const { selectedTabIndex, setSelectedTabIndex } =
    useContext<SelectedContextType>(SelectedContext);
  const navigate = useNavigate();

  useEffect(() => {
    if (
      isMyProfileFetchSuccess &&
      myProfile.data.role &&
      myProfile.data.role.permissionLevel === PermissionLevel.SUPER
    ) {
      setIsSuperAuthorized(true);
      setIsBasicAuthorized(false);
    } else if (isMyProfileFetchError) {
      setIsBasicAuthorized(true);
    }
  }, [isMyProfileFetchSuccess, isMyProfileFetchError]);

  const onEmployeeListSelected = () => {
    setSelectedTabIndex(0);
    navigate(RouteConstants.employee);
  };

  const onJobOpeningsSelected = () => {
    setSelectedTabIndex(1);
    if (isSuperAuthorized) navigate(RouteConstants.opening);
    else navigate(RouteConstants.publicOpening);
  };

  const onApplicationsSelected = () => {
    setSelectedTabIndex(2);
    if (isSuperAuthorized) navigate(RouteConstants.application);
  };

  const onReferralsListSelected = () => {
    setSelectedTabIndex(3);
    navigate(RouteConstants.referral);
  };

  return (
    <div className='sidebar'>
      <div className='sidebar-items'>
        {!isBasicAuthorized && (
          <SideBarButton
            isSelected={selectedTabIndex === 0}
            onClick={onEmployeeListSelected}
            imgIcon='/assets/icons/employees.png'
            headerText='Employee List'
          />
        )}
        <SideBarButton
          isSelected={selectedTabIndex === 1}
          onClick={onJobOpeningsSelected}
          imgIcon='/assets/icons/opening.png'
          headerText='Job Openings'
        />

        {isSuperAuthorized && (
          <SideBarButton
            isSelected={selectedTabIndex === 2}
            onClick={onApplicationsSelected}
            imgIcon='/assets/icons/application.png'
            headerText='Applications'
          />
        )}
        {!isBasicAuthorized && (
          <SideBarButton
            isSelected={selectedTabIndex === 3}
            onClick={onReferralsListSelected}
            imgIcon='/assets/icons/referral.png'
            headerText='Referrals'
          />
        )}
      </div>
    </div>
  );
};

export default Sidebar;
