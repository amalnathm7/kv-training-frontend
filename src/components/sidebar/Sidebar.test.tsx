import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Sidebar from './Sidebar';
import { Router } from 'react-router-dom';

describe('Sidebar Props Test', () => {
  test('If Sidebar rendered correctly', () => {
    const element = render(
      <Router location={''} navigator={undefined}>
        <Sidebar />
      </Router>
    );

    expect(element).toMatchSnapshot();
  });
});
