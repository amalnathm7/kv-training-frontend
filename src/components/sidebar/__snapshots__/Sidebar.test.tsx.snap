// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Props Test If Sidebar rendered correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sidebar"
      >
        <div
          class="sidebar-items"
        >
          <div
            class="item container"
            data-testid="sidebar-button-test"
          >
            <div
              class="icon container"
            >
              <img
                alt="Icon"
                class="filtered"
                src="/assets/icons/employees.svg"
                style="height: 15px; width: 15px;"
              />
            </div>
            <h3>
              Employee List
            </h3>
          </div>
          <div
            class="item container"
            data-testid="sidebar-button-test"
          >
            <div
              class="icon container"
            >
              <img
                alt="Icon"
                class="filtered"
                src="/assets/icons/job_icon.png"
                style="height: 15px; width: 15px;"
              />
            </div>
            <h3>
              Job Openings
            </h3>
          </div>
          <div
            class="item container"
            data-testid="sidebar-button-test"
          >
            <div
              class="icon container"
            >
              <img
                alt="Icon"
                class="filtered"
                src="/assets/icons/my_referrals.png"
                style="height: 15px; width: 15px;"
              />
            </div>
            <h3>
              My Referrals
            </h3>
          </div>
          <div
            class="item container"
            data-testid="sidebar-button-test"
          >
            <div
              class="icon container"
            >
              <img
                alt="Icon"
                class="filtered"
                src="/assets/icons/all_referrals.png"
                style="height: 15px; width: 15px;"
              />
            </div>
            <h3>
              All Referrals
            </h3>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sidebar"
    >
      <div
        class="sidebar-items"
      >
        <div
          class="item container"
          data-testid="sidebar-button-test"
        >
          <div
            class="icon container"
          >
            <img
              alt="Icon"
              class="filtered"
              src="/assets/icons/employees.svg"
              style="height: 15px; width: 15px;"
            />
          </div>
          <h3>
            Employee List
          </h3>
        </div>
        <div
          class="item container"
          data-testid="sidebar-button-test"
        >
          <div
            class="icon container"
          >
            <img
              alt="Icon"
              class="filtered"
              src="/assets/icons/job_icon.png"
              style="height: 15px; width: 15px;"
            />
          </div>
          <h3>
            Job Openings
          </h3>
        </div>
        <div
          class="item container"
          data-testid="sidebar-button-test"
        >
          <div
            class="icon container"
          >
            <img
              alt="Icon"
              class="filtered"
              src="/assets/icons/my_referrals.png"
              style="height: 15px; width: 15px;"
            />
          </div>
          <h3>
            My Referrals
          </h3>
        </div>
        <div
          class="item container"
          data-testid="sidebar-button-test"
        >
          <div
            class="icon container"
          >
            <img
              alt="Icon"
              class="filtered"
              src="/assets/icons/all_referrals.png"
              style="height: 15px; width: 15px;"
            />
          </div>
          <h3>
            All Referrals
          </h3>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
